src/runtime/job.h

```diff
@@ -19,15 +19,34 @@
 #define JOB_H
 
 #include <memory>
+#include <string>
 
 struct Database;
 struct Runtime;
 
+struct ResourceBudget {
+  double percentage;
+  uint64_t fixed;
```

Comment:
It seems like `percentage` and `fixed` have a specific rule that decides which one is actually valid (which is that `fixed` has priority if it's non-zero). Could you either document that, or change this from a `struct` to a `class` so that the members are private?

Reply:
Done.